### PR TITLE
tests: add unit tests for AugurUUID and subclasses

### LINKED ISSUE #3744: Unit test AugurUUID and its subclasses
These look like very well-built and isolated classes that would be easy to test

making this issue in case someone wants to start on it.

### PR DESCRIPTION
**Pull request by Inengs**
**Description**
- Adds unit tests for AugurUUID and its subclasses

Tests cover:
- Initialization with 16 zero bytes
- Github and Gitlab produce different UUIDs for same user ID
- Equality operations
- Platform ID set correctly for each subclass
- toUUID() returns valid uuid.UUID object
- Error handling for invalid values and out of range indexes.

This PR fixes #3744

**Notes for Reviewers**

**Signed commits**
- [x] Yes, I signed my commits.

### CODE REVIEW

**Comment by MoralCode** [comment]
yeah ignore bandit there

Can you add this test folder/file to the pytest section of `pyproject.toml` so that it gets run along with the rest of the test cases?

**Comment by Inengs** [comment]
Alright I will do that as soon as possible, thank you

**Comment by Inengs** [comment]
Hi, I have added the test file to the pyproject.toml 

please let me know if there is anything else needed, thank you!

File: tests/test_tasks/test_task_utilities/test_util/test_augur_uuid.py [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+def test_github_uuid_set_user():
+    uid = GithubUUID()
+    uid["user"] = 12345
+    assert uid["user"] == 12345
+
+# checks the that you can store a value in the user field
+def test_gitlab_uuid_set_user():
+    uid = GitlabUUID()
+    uid["user"] = 99999
+    assert uid["user"] == 99999
```
Comment (MoralCode) [inline tests/test_tasks/test_task_utilities/test_util/test_augur_uuid.py]: this test is probably redundant with other tests

File: tests/test_tasks/test_task_utilities/test_util/test_augur_uuid.py [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+    assert uid["platform"] == 1
+
+# checks that gitlabUUID sets its platform number to 2
+def test_gitlab_uuid_platform_is_2():
+    uid = GitlabUUID()
+    assert uid["platform"] == 2
+
+def test_unresolvable_uuid_platform_is_0():
+    uid = UnresolvableUUID()
+    assert uid["platform"] == 0
```
Comment (MoralCode) [inline tests/test_tasks/test_task_utilities/test_util/test_augur_uuid.py]: this test is pretty trivial and therefore not really that helpful

**Review by MoralCode — CHANGES_REQUESTED** [review]
Id like to see fewer test cases that just assert that each type's identifier is correct and more tests that exercise the various edge cases and code branches within the logic of AugurUUID (even if you use a GitHubUUID instance for doing the test)

**Comment by Inengs** [comment]
Alright, I will make those changes as soon as possible..

**Comment by Inengs** [comment]
Hi, I have updated the tests based on your feedback

please let me know if there is anything else needed, thank you!

File: tests/test_tasks/test_task_utilities/test_util/test_augur_uuid.py [hunk 3]
```diff
@@ -71,16 +127,56 @@ def test_inequality():
...
+
+# checks the minimum boundary
+def test_user_field_zero_value():
+    uid = GithubUUID()
+    uid["user"] = 0
+    assert uid["user"] == 0
+
+def test_len_returns_16():
+    uid = AugurUUID()
+    assert len(uid) == 16
```
Comment (MoralCode) [inline tests/test_tasks/test_task_utilities/test_util/test_augur_uuid.py]: these tests arent really that helpful

File: tests/test_tasks/test_task_utilities/test_util/test_augur_uuid.py [hunk 4]
```diff
@@ -71,16 +127,56 @@ def test_inequality():
...
+    result = str(uid)
+
+    assert "user" in result
+    assert "platform" in result
+
+def test_iteration_over_bytes():
+    uid = AugurUUID()
+    bytes_list = list(uid)
+
+    assert len(bytes_list) == 16
```
Comment (MoralCode) [inline tests/test_tasks/test_task_utilities/test_util/test_augur_uuid.py]: this is effectively the same as the len() test

**Comment by Inengs** [comment]
Hi, I have removed the redundant tests.

please let me know if there is anything needed to add, thank you!
